onboard/catkin_ws/src/task_planning/scripts/move_tasks.py

```diff
@@ -0,0 +1,115 @@
+from task import Task
+from geometry_msgs.msg import Pose, Quaternion, Twist, Point, Vector3
+from nav_msgs.msg import Odometry
+from tf.transformations import quaternion_from_euler
+import task_utils
+import rospy
+
+
+class MoveToPoseGlobalTask(Task):
+    """Move to pose given in global coordinates."""
+
+    def __init__(self, x, y, z, roll, pitch, yaw):
+        super(MoveToPoseGlobalTask, self).__init__()
+
+        self.desired_pose = Pose()
+        self.desired_pose.position = Point(x=x, y=y, z=z)
+        self.desired_pose.orientation = Quaternion(*quaternion_from_euler(roll, pitch, yaw))
+        self.desired_twist = Twist()  # default 0 angular and linear velocity
+
+    def _on_task_run(self):
+        self.publish_desired_pose_global(self.desired_pose)
+        at_desired_pose_vel = task_utils.stopped_at_pose(self.state.pose.pose, self.desired_pose, self.state.twist.twist)
+        if at_desired_pose_vel:
+            self.finish()
+
+
+class MoveToPoseLocalTask(MoveToPoseGlobalTask):
+    """Move to pose given in local coordinates."""
+    # this can be shortened, no reason to override _on_task_run
```

Comment:
Yes, shorten it if possible, remove the on_task_run method. You will need to rename tranformed_pose to desired_pose in on_task_start I think (make sure that is ok for restarting the task). You might need to override restart, as mentioned in another comment.